### PR TITLE
Revised documentation

### PR DESCRIPTION
**Pull request by tfalders**
Addresses SWDEV-291340.

This change brings hipSOLVER's documentation more in line with that of rocSOLVER. A lot of info has been moved out of the github readme and into relevant documentation pages, and an API listing has been added.

### CODE REVIEW

File: docs/source/userguide_install.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+======================================================
+
+The repository contains source code for client programs that serve as tests, benchmarks, and samples. Client source code can be found in the clients subdirectory.
+
+Dependencies (only necessary for hipSOLVER clients)
+----------------------------------------------------
+
+The hipSOLVER samples have no external dependencies, but our unit test and benchmarking applications do. These clients introduce the following dependencies:
+
+1. `boost <https://www.boost.org/>`_
```
Comment (jzuniga-amd) [inline docs/source/userguide_install.rst:69]: do we still need boost in hip solver?

**Review by jzuniga-amd — APPROVED** [review]
Later on, we can decide on whether to add descriptions to the different functions and types in the API, but this looks good to me for the first version. Just one comment below...

File: docs/source/userguide_install.rst (hunk 1 above)
Comment (tfalders) [reply docs/source/userguide_install.rst:69]: Good point! I'll take it out.

File: README.md [hunk 2]
```diff
@@ -1,73 +1,31 @@
...
-* `./install.sh -ic` -- build library, build clients (tests, benchmarks, and samples), and install.
+## Quickstart Build
 
-To see more options, use the help option of the install script.
-* `./install.sh -h`
+To download the hipSOLVER source code, clone this repository with the command:
 
-### Build & Install Library Manually
-For a standard library installation, follow these steps:
+    git clone git@github.com:ROCmSoftwarePlatform/hipSOLVER.git
```
Comment (cgmb) [inline README.md:12]: Cloning over SSH requires creating a GitHub account and registering your SSH public key. I'd swap that out for `git clone https://github.com/ROCmSoftwarePlatform/hipSOLVER.git`, which can be done without any additional setup.

File: README.md [hunk 3]
```diff
@@ -1,73 +1,31 @@
...
-```sh
-CXX=/opt/rocm/bin/hcc cmake -DBUILD_CLIENTS_TESTS=ON -DBUILD_CLIENTS_BENCHMARKS=ON [HIPSOLVER_SOURCE]
-make -j$(nproc)
-sudo make install   # sudo required if installing into system directory such as /opt/rocm
-```
+For more information on building and installing hipSOLVER, see the [hipSOLVER install guide](https://hipsolver.readthedocs.io/en/latest/userguide_install.html)
 
 ## Using the hipSOLVER Interface
-The hipSOLVER interface is compatible with rocSOLVER and cuBLAS-v11 APIs. Porting a CUDA application that originally calls the cuSOLVER API to an application calling the hipSOLVER API should be relatively straightforward. For example, the hipSOLVER SGETRF interface is
+The hipSOLVER interface is compatible with rocSOLVER and cuBLAS-v11 APIs. Porting a CUDA application that originally calls the cuSOLVER API to an application calling the hipSOLVER API should be relatively straightforward (with some caveats; see [here](https://hipsolver.readthedocs.io/en/latest/userguide_intro.html#differences-with-the-cusolver-api)). For example, the hipSOLVER SGEQRF interface is
```
Comment (cgmb) [inline README.md:24]: If possible, try to ensure the link text is a useful description that can stand on its own. Perhaps, 'see the [List of Differences from the cuSOLVER API](https://hipsolver.readthedocs.io/en/latest/userguide_intro.html#differences-with-the-cusolver-api)'. The article, [What’s wrong with using "Click Here" links?](https://webaccess.berkeley.edu/ask-pecan/click-here) explains why in a bit more detail.

Also, I'm the worst about this sort of thing, but 'relatively straightforward' is weak and vague writing. Relative to what? Drop 'relative' and it's a better sentence.

**Review by cgmb — COMMENTED** [review]
A couple of minor comments.

File: README.md (hunk 3 above)
Comment (tfalders) [reply README.md:24]: The `relative` part is copied from hipBLAS. I think the statement is too strong with it removed, so perhaps I will replace it with `fairly`.

**Review by cgmb — APPROVED** [review]

**Comment by tfalders** [comment]
There are some failures on CI, but this PR doesn't touch any code so they ought to be unrelated. I'll go ahead and merge.

**Pull request merged**